examples/button.rs

```diff
@@ -0,0 +1,591 @@
+                        }, 
+                        BUTTON_THEME
+                    )).with_children(| button | {
+                        button.spawn_bundle(TextBundle::from_section(
+                            Icon::Delete.to_string(),
+                            TextStyle {
+                                font: asset_server.load(MATERIAL_FONT).into(),
+                                font_size: BUTTON_FONT_SIZE,
+                                color: COLOR_TEXT,
+                            }
+                        ).with_style(Style {
+                            margin: UiRect::all(Val::Px(10.0)),
+                            ..default()
+                        }));
+                        button.spawn_bundle(TextBundle::from_section(
+                            "Delete",
+                            TextStyle {
+                                font: asset_server.load(TEXT_FONT).into(),
+                                font_size: BUTTON_FONT_SIZE,
+                                color: COLOR_TEXT,
+                            }
+                        ));
+                    });
+                });
+
+                // Title text
+                content.spawn_bundle(TextBundle::from_section(
+                    "Disabled Buttons", 
+                    TextStyle {
+                        font: asset_server.load(H1_FONT).into(),
+                        font_size: H1_FONT_SIZE,
+                        color: COLOR_TEXT,
+                    }
+                ).with_style(Style {
+                    margin: UiRect::new(Val::Undefined, Val::Undefined, Val::Px(15.0), Val::Undefined),
+                   ..default()
+               }));
+                // Paragraph
+                content.spawn_bundle(TextBundle::from_section(
+                    "Buttons can be either enabled or disabled. \
+                            When they are disabled, they should look the part, \
+                            and not be able to be triggered by the user.",
+                    TextStyle {
+                        font: asset_server.load(TEXT_FONT).into(),
+                        font_size: TEXT_FONT_SIZE,
+                        color: COLOR_TEXT,
+                    }
+                ));
+                
+                // Example Showcase
+                content.spawn_bundle(NodeBundle {
+                    style: Style {
+                        size: Size::new(Val::Percent(100.0), Val::Px(80.0)),
+                        margin: UiRect::new(Val::Undefined, Val::Undefined, Val::Px(10.0), Val::Px(10.0)),
+                        flex_direction: FlexDirection::Row,
+                        align_items: AlignItems::Center,
+                        justify_content: JustifyContent::Center,
+                        
+                        ..default()
+                    },
+                    color: COLOR_CONTENT_EXAMPLE.into(),
+                    ..default()
+                }).with_children(| example_showcase | {
```

Comment:
At some points you might be able to reduce the boilerplate by using loops to spawn the UI nodes. This can work if you have only a few parts (like inner text) that change and many common parts.

For example
```rust
for text in ["Ok", "Submit", "Enable Wifi"] {
   // Spawn bundles with changing text
}
```

This can be extended to be an array of tuples if you have multiple changing components. Not sure if it would work well here, but might be worth a try :)

Reply:
Good idea, this is basically what I wanted to achieve with the `WidgetCommands`-idea, but this is much less intrusive :)